src/vos/evtree.c

```diff
@@ -2293,61 +2317,41 @@ evt_insert(daos_handle_t toh, const struct evt_entry_in *entry,
 		return rc;
 
 	if (ent_array->ea_ent_nr == 1) {
-		if (entry->ei_rect.rc_minor_epc == EVT_MINOR_EPC_MAX) {
-			/** Special case.   This is an overlapping delete record
-			 *  which can happen when there are minor epochs
-			 *  involved.   Rather than rejecting, insert prefix
-			 *  and/or suffix extents.
```

Comment:
Why this is removed? No overlapping removal records now?

Reply:
Removal records are inserted by doing a visible find and then inserting a record for each visible extent.  They should never overlap.  I think this code is there from the original implementation of insert for removal records.